connutils.go

```diff
@@ -45,8 +45,60 @@ type connConfig struct {
 	database           string
 	connectTimeout     time.Duration
 	waitUntilAvailable time.Duration
+	tlsCAData          []byte
+	tlsSecurity        string
 	serverSettings     map[string]string
-	tlsConfig          *tls.Config
+}
+
+func (c *connConfig) tlsConfig() (*tls.Config, error) {
+	var roots *x509.CertPool
+	if len(c.tlsCAData) != 0 {
+		roots = x509.NewCertPool()
+		if ok := roots.AppendCertsFromPEM(c.tlsCAData); !ok {
+			return nil, errors.New("invalid certificate data")
+		}
+	} else {
+		var err error
+		roots, err = getSystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	tlsConfig := &tls.Config{
+		RootCAs:    roots,
+		NextProtos: []string{"edgedb-binary"},
+	}
+
+	security, err := getEnvVarSetting(
+		"EDGEDB_CLIENT_SECURITY", "default", "default", "insecure_dev_mode",
+		"strict")
```

Comment:
`strict` should enable `tls_security=strict` (or error if it's set to a conflicting value).